lib/benchee/formatters/console/memory.ex

```diff
@@ -0,0 +1,199 @@
+    ]
+  end
+
+  defp scenario_reports(scenarios, units, label_width, hide_statistics) do
+    Enum.map(scenarios, fn scenario ->
+      format_scenario(scenario, units, label_width, hide_statistics)
+    end)
+  end
+
+  @spec format_scenario(Scenario.t(), unit_per_statistic, integer, boolean) :: String.t()
+  defp format_scenario(scenario, %{memory: memory_unit}, label_width, false) do
+    %Scenario{
+      name: name,
+      memory_usage_statistics: %Statistics{
+        average: average,
+        std_dev_ratio: std_dev_ratio,
+        median: median,
+        percentiles: %{99 => percentile_99}
+      }
+    } = scenario
+
+    "~*ts~*ts~*ts~*ts~*ts\n"
+    |> :io_lib.format([
+      -label_width,
+      name,
+      @average_width,
+      Helpers.run_time_out(average, memory_unit),
+      @deviation_width,
+      Helpers.deviation_out(std_dev_ratio),
+      @median_width,
+      Helpers.run_time_out(median, memory_unit),
+      @percentile_width,
+      Helpers.run_time_out(percentile_99, memory_unit)
+    ])
+    |> to_string
+  end
+
+  defp format_scenario(scenario, %{memory: memory_unit}, label_width, true) do
+    %Scenario{
+      name: name,
+      memory_usage_statistics: %Statistics{
+        average: average
+      }
+    } = scenario
+
+    "~*ts~*ts\n"
+    |> :io_lib.format([
+      -label_width,
+      name,
+      @average_width,
+      Helpers.run_time_out(average, memory_unit)
+    ])
+    |> to_string
+  end
+
+  @spec comparison_report([Scenario.t()], unit_per_statistic, integer, map, boolean) :: [
+          String.t()
+        ]
+  defp comparison_report(scenarios, units, label_width, config, hide_statistics)
+
+  # No need for a comparison when only one benchmark was run
+  defp comparison_report([_scenario], _, _, _, _), do: []
+  defp comparison_report(_, _, _, %{comparison: false}, _), do: []
+  defp comparison_report(_, _, _, _, true), do: []
+
+  defp comparison_report([scenario | other_scenarios], units, label_width, _, _) do
+    [
+      Helpers.descriptor("Comparison"),
+      reference_report(scenario, units, label_width) |
+      comparisons(scenario, units, label_width, other_scenarios)
+    ]
+  end
+
+  defp reference_report(scenario, %{memory: memory_unit}, label_width) do
+    %Scenario{
+      name: name,
+      memory_usage_statistics: %Statistics{median: median}
+    } = scenario
+
+    "~*s~*s\n"
+    |> :io_lib.format([
+      -label_width,
+      name,
+      @median_width,
+      Helpers.run_time_out(median, memory_unit)
+    ])
+    |> to_string
+  end
+
+  @spec comparisons(Scenario.t(), unit_per_statistic, integer, [Scenario.t()]) :: [String.t()]
+  defp comparisons(scenario, units, label_width, scenarios_to_compare) do
+    %Scenario{memory_usage_statistics: reference_stats} = scenario
+
+    Enum.map(scenarios_to_compare, fn scenario = %Scenario{memory_usage_statistics: job_stats} ->
+      slower =
+        if job_stats.median == 0 || reference_stats.median == 0 do
+          0.0
```

Comment:
No, this is really `0.0`. I just needed it to be a float like all the other values so it didn't blow up when we put that value into `:io_lib.format`, and also so we don't have any accidental "divide by 0" errors.

I'll pull it out into a separate function and we'll see what we think about it. We can always put it back together if it ends up feeling weird.

Reply:
But if the median is 0 - it's infinitely faster or slower so I think `N/A` makes sense doesn't it? Because 0 times faster or slower sounds weird? :thinking:

Reply:
Hmmm, well, this is just for memory usage, not for run times. But the point is still solid - I guess we could omit the "x times more memory" all together if it's `0.0` or `N/A` or whatever.